whelktool/scripts/cleanups/2019/10/lxl-2594-fix-faulty-encodingLevels/set-correct-bib-encodingLevel-value.groovy

```diff
@@ -0,0 +1,33 @@
+/*
+ * This removes auth and hold specific values of encodingLevel
+ * and set correct valid bib level
+ *
+ * See LXL-2594 for more info.
+ *
+ */
+
+PrintWriter failedBibIDs = getReportWriter("failed-to-update-bibIDs")
+PrintWriter scheduledForUpdate = getReportWriter("scheduled-for-update")
+VALID_BIB_ENCODING_LEVELS = ["marc:AbbreviatedLevel",
+                             "marc:CoreLevel",
+                             "marc:FullLevel",
+                             "marc:FullLevelMaterialNotExamined",
+                             "marc:LessThanFullLevelMaterialNotExamined",
+                             "marc:MinimalLevel",
+                             "marc:PartialPreliminaryLevel",
+                             "marc:PrepublicationLevel"]
+
+selectBySqlWhere("""
+        collection = 'bib' AND data#>>'{@graph,0,encodingLevel}' IS NOT NULL
+    """) { data ->
+
+    def (record, instance, work) = data.graph
+
+    if (!VALID_BIB_ENCODING_LEVELS.contains(record['encodingLevel'])) {
+        record.put('encodingLevel', VALID_BIB_ENCODING_LEVELS[0])
```

Comment:
Could extract first value as an `ABBR_LEVEL` constant, use that in the list and use `as Set` on that (for clarity and an admittedly negligible speedup).